test/Libraries/Microsoft.Extensions.AI.OpenAI.Tests/OpenAISpeechToTextClientTests.cs

```diff
@@ -0,0 +1,388 @@
+    }
+
+    [Fact]
+    public async Task NonStronglyTypedOptions_AllSent()
+    {
+        const string Input = """
+                {
+                    "model": "whisper-1",
+                    "prompt":"Hide any bad words with ",
+                    "temperature": 0.5,
+                    "response_format": "vtt",
+                    "timestamp_granularities[]": ["word","segment"]
+                }
+                """;
+
+        const string Output = """
+                {
+                    "text":"I finally got back to the gym the other day."
+                }
+                """;
+
+        using VerbatimMultiPartHttpHandler handler = new(Input, Output);
+        using HttpClient httpClient = new(handler);
+        using ISpeechToTextClient client = CreateSpeechToTextClient(httpClient, "whisper-1");
+
+        using var fileStream = GetAudioStream("audio001.mp3");
+        Assert.NotNull(await client.GetResponseAsync(fileStream, new()
+        {
+            AdditionalProperties = new()
+            {
+                ["SpeechLanguage"] = "pt",
+                ["Temperature"] = 0.5f,
+                ["TimestampGranularities"] = AudioTimestampGranularities.Segment | AudioTimestampGranularities.Word,
+                ["Prompt"] = "Hide any bad words with ***",
+                ["ResponseFormat"] = AudioTranscriptionFormat.Vtt,
+            },
+        }));
+    }
+
+    [Fact]
+    public async Task StronglyTypedOptions_AllSent()
+    {
+        const string Input = """
+                {
+                    "model": "whisper-1",
+                    "language": "pt"
+                }
+                """;
+
+        const string Output = """
+                {
+                    "text":"I finally got back to the gym the other day."
+                }
+                """;
+
+        using VerbatimMultiPartHttpHandler handler = new(Input, Output);
+        using HttpClient httpClient = new(handler);
+        using ISpeechToTextClient client = CreateSpeechToTextClient(httpClient, "whisper-1");
+
+        using var fileStream = GetAudioStream("audio001.mp3");
+        Assert.NotNull(await client.GetResponseAsync(fileStream, new()
+        {
+            SpeechLanguage = "pt",
+        }));
+    }
+
+    private static Stream GetAudioStream(string fileName)
+#pragma warning restore S125 // Sections of code should not be commented out
+    {
+        using Stream? s = typeof(OpenAISpeechToTextClientTests).Assembly.GetManifestResourceStream($"Microsoft.Extensions.AI.Resources.{fileName}");
+        Assert.NotNull(s);
+        MemoryStream ms = new();
+        s.CopyTo(ms);
+
+        ms.Position = 0;
+        return ms;
+    }
+
+    private static ISpeechToTextClient CreateSpeechToTextClient(HttpClient httpClient, string modelId) =>
+        new OpenAIClient(new ApiKeyCredential("apikey"), new OpenAIClientOptions { Transport = new HttpClientPipelineTransport(httpClient) })
+        .AsSpeechToTextClient(modelId);
+}
```

Comment:
Can we avoid adding these large wav files? We especially don't want to add them multiple times.

Once merged, they'll be in the repo's history forever.

Reply:
Yep, dropped using the files in the UnitTest and changed to mp3 just for the integration.